.buildkite/scripts/aws_cloud_tests.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+
+REPO_DIR=$(pwd)
+
+teardown() {
+  # reset the directory to the root of the project
+  cd $REPO_DIR
+  # Teardown resources after using them
+  echo "~~~ Terraform Cleanup"
+  tf_cleanup "${MODULE_DIR}"              #TODO: move all docker-compose files from the .ci to .buildkite folder before switching to BK
+
+  echo "~~~ Docker Compose Cleanup"
+  docker-compose -f .ci/jobs/docker-compose.yml down -v         #TODO: move all docker-compose files from the .ci to .buildkite folder before switching to BK
+}
+
+tf_cleanup() {
+  DIRECTORY=${1:-.}
+
+  for tfstate in $(find $DIRECTORY -name terraform.tfstate); do
+    cd $(dirname $tfstate)
+    terraform init
+    if ! terraform destroy -auto-approve; then
+        echo "+++ Failed to Terraform destroy the resources"
+    fi
+    cd -
+  done
+}
+
+trap 'teardown' EXIT
+
+# Prepare the cloud resources using Terraform
+#startCloudTestEnv "${MODULE_DIR}"
+echo "~~~ Loading creds"
+set +o xtrace
+export AWS_ACCESS_KEY_ID=$BEATS_AWS_ACCESS_KEY
+export AWS_SECRET_ACCESS_KEY=$BEATS_AWS_SECRET_KEY
+export TEST_TAGS="${TEST_TAGS:+$TEST_TAGS,}aws"
+set -o xtrace
+
+echo "~~~ Run docker-compose services for emulated cloud env"
+docker-compose -f .ci/jobs/docker-compose.yml up -d        #TODO: move all docker-compose files from the .ci to .buildkite folder before switching to BK
+echo "~~~ Initialize TF cloud resources"
+cd "$MODULE_DIR"
+export TF_VAR_BRANCH=$(echo "${BUILDKITE_BRANCH}" | tr '[:upper:]' '[:lower:]' | sed 's/[^a-z0-9-]/-/g')
+export TF_VAR_BUILD_ID="${BUILDKITE_BUILD_ID}"
+export TF_VAR_CREATED_DATE=$(date +%s)
+export TF_VAR_ENVIRONMENT="ci"
+export TF_VAR_REPO="${REPO}"
+terraform init && terraform apply -auto-approve
+cd -
+
+# Run tests
+echo "~~~ Run Cloud Tests for $BEATS_PROJECT_NAME"
+cd "${BEATS_PROJECT_NAME}"
+mage build test goIntegTest
```

Comment:
AFAICS the only difference between this script and the existing one `.buildkite/scripts/cloud_tests.sh` is the actual command to execute:

```
$ diff .buildkite/scripts/cloud_tests.sh .buildkite/scripts/aws_cloud_tests.sh 
57c57
< mage build test
---
> mage build test goIntegTest
```

can we instead reuse the existing `.buildkite/scripts/cloud_tests.sh` instead? Ideally the script shouldn't be running `mage` but it should renamed to e.g. `prepare_cloud_tests.sh` and used like:

```
command:
  source .buildkite/scripts/prepare_cloud_tests.sh
  cd x-pack/filbeat
  mage build test goIntegTest
```

I understand that this script is currently used in `x-pack/pipeline.xpack.metricbeat.yml` and `x-pack/pipeline.xpack.filebeat.yml` so maybe you don't want to change both in **this PR**, but for now you could instead allow passing a command to run e.g. by replacing lines 54-57 above with

```
# TODO remove this conditional entirely and change the way we use this script, by sourcing it
# and explicitly invoking cd ... && mage ... within the step command
if [ "$#" -eq 0 ]; then
    # run default tests if no parameters passed
   echo "~~~ Run Cloud Tests [mage build test] for $BEATS_PROJECT_NAME"
  cd "${BEATS_PROJECT_NAME}"
  mage build test
else
  echo "~~~ Running Cloud Tests [$1] for $BEATS_PROJECT_NAME"
  cd "${BEATS_PROJECT_NAME}"
  eval "$1"
fi
```

which should be backwards compatible with the other pipeline, but can be used here like:

```
        command: |
          set -euo pipefail
          # defines the MODULE env var based on what's changed in a PR
          source .buildkite/scripts/changesets.sh
          defineModuleFromTheChangeSet x-pack/filebeat
          echo "~~~ Running tests with env var MODULE=$$MODULE"
          .buildkite/scripts/cloud_tests.sh "mage build test goIntegTest"
```

Reply:
As agreed - will updated both x-pack/pipeline.xpack.metricbeat.yml and x-pack/pipeline.xpack.filebeat.yml to use same cloud env setup script and move cd <beat-name> && <test command> to pipeline step explicitly.